client/my-sites/backup/backup-date-picker/test/index.jsx

```diff
@@ -104,127 +110,117 @@ describe( 'BackupDatePicker', () => {
-			<BackupDatePicker selectedDate={ december4Of2020 } onDateChange={ () => {} } />
-		);
-
-		const nextDate = picker.find(
-			'.backup-date-picker__select-date--next .backup-date-picker__display-date'
-		);
+		render( <BackupDatePicker selectedDate={ december4Of2020 } onDateChange={ () => {} } /> );
 
 		const expectedText = moment( december4Of2020 ).add( 1, 'day' ).format( 'MMM D' );
-		expect( nextDate.text() ).toEqual( expectedText );
+
+		const nextDateText = screen.getByText( expectedText );
+		expect( nextDateText ).toBeVisible();
+		expect( nextDateText ).toHaveClass( 'backup-date-picker__display-date' );
+		expect( nextDateText.parentElement ).toHaveClass( 'backup-date-picker__next-date-link' );
 	} );
 
 	test( "Does not show the next date as 'Yesterday' even if it is yesterday", () => {
-		const today = moment();
-
-		const picker = shallow( <BackupDatePicker selectedDate={ today } onDateChange={ () => {} } /> );
+		const twoDaysBefore = moment().subtract( 2, 'days' );
+		render( <BackupDatePicker selectedDate={ twoDaysBefore } onDateChange={ () => {} } /> );
 
-		const previousDate = picker.find(
-			'.backup-date-picker__select-date--next .backup-date-picker__display-date'
-		);
-		expect( previousDate.text() ).not.toEqual( 'Yesterday' );
+		expect( screen.queryByText( 'Yesterday' ) ).not.toBeInTheDocument();
 	} );
 
 	test( "Does not show the next date as 'Today', even if it is today", () => {
 		const yesterday = moment().subtract( 1, 'day' );
 
-		const picker = shallow(
-			<BackupDatePicker selectedDate={ yesterday } onDateChange={ () => {} } />
-		);
+		render( <BackupDatePicker selectedDate={ yesterday } onDateChange={ () => {} } /> );
 
-		const previousDate = picker.find(
-			'.backup-date-picker__select-date--next .backup-date-picker__display-date'
-		);
-		expect( previousDate.text() ).not.toEqual( 'Today' );
+		expect( screen.queryByText( 'Today' ) ).not.toBeInTheDocument();
 	} );
 
 	test( "Shows month, date, and year for the next date if it's not the same year as today", () => {
 		const twoYearsAgo = moment().subtract( 2, 'years' );
 
-		const picker = shallow(
-			<BackupDatePicker selectedDate={ twoYearsAgo } onDateChange={ () => {} } />
-		);
-
-		const previousDate = picker.find(
-			'.backup-date-picker__select-date--next .backup-date-picker__display-date'
-		);
+		render( <BackupDatePicker selectedDate={ twoYearsAgo } onDateChange={ () => {} } /> );
 
 		const expectedText = moment( twoYearsAgo ).add( 1, 'day' ).format( 'MMM D, YYYY' );
-		expect( previousDate.text() ).toEqual( expectedText );
+
+		const nextDateText = screen.getByText( expectedText );
+		expect( nextDateText ).toBeVisible();
+		expect( nextDateText ).toHaveClass( 'backup-date-picker__display-date' );
+		expect( nextDateText.parentElement ).toHaveClass( 'backup-date-picker__next-date-link' );
 	} );
 
 	// --- NAVIGATION ---
 
 	test( 'Navigates backward when the previous date is clicked', () =>
 		new Promise( ( done ) => {
```

Comment:
The current way of writing those async tests is not ideal, since the order is a bit confusing - the assertions are at the top, and the rendering is at the bottom. To improve this, I'd suggest to:

* Remove the promise
* Make the test function `async`.
* Add an `await` before any `user.*` event trigger, so the assertions would be run after it fully finishes execution.
* Move the assertions to the bottom of the tests.

This applies to the rest of the same style tests that directly rely on a `Promise`.

What do you think?

Reply:
I have updated to use the async function instead of a promise. Thanks